packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -235,9 +256,19 @@ export class EVMAAWallet<B extends EVMBlockchainIncludingTestnet = EVMBlockchain
     getSigner<Type extends SignerType>(type: Type): SignerMap[Type] {
         switch (type) {
             case "viem": {
+                //zerodev returns a type where "account" could be undefined, and "walletClient" requires it to be defined
+                // and this hack only changes that type. So, the change is:
+                // KernelSmartAccount<EntryPoint, HttpTransport, TChain> | undefined => KernelSmartAccount<EntryPoint, HttpTransport, TChain>
+                const kernelClientHackType = this.kernelClient as KernelAccountClient<
+                    EntryPoint,
+                    HttpTransport,
+                    TChain,
+                    KernelSmartAccount<EntryPoint, HttpTransport, TChain>
+                >;
```

Comment:
maybe we should do some validation here then before changing it to a more "dangerous" type
What do you think about
```suggestion
                if (this.kernelClient.account == null) {
                    throw new Error(...)
                }
                const kernelClientHackType = this.kernelClient as KernelAccountClient<
                    EntryPoint,
                    HttpTransport,
                    TChain,
                    KernelSmartAccount<EntryPoint, HttpTransport, TChain>
                >;
```